src/typ.rs

```diff
@@ -1626,11 +1630,24 @@ pub fn infer_module(
     // Remove private and imported types and values to create the public interface
     env.module_types
         .retain(|_, info| info.public && &info.module == module_name);
-    env.module_values.retain(|_, info| info.public);
+
+    // Move the hashmap of module values out of the env struct, then iterate over it.
+    // Create warnings for uncalled private functions
+    // Finally, keep only the public values.
+    let mut module_values = std::mem::replace(&mut env.module_values, HashMap::new());
```

Comment:
Couldn't we keep this hashmap on env as before?

Reply:
We probably can but this seemed like an easier way to make the borrow checker happy. Trying to add warnings to the env inside the retain was giving me the old multiple mutable references. I'm sure I can wrestle it into submission if you'd rather keep it on the env.